include/platform/mir/graphics/platform.h

```diff
@@ -71,11 +179,279 @@ class RenderingPlatform
     /**
      * Creates the buffer allocator subsystem.
      */
-    virtual UniqueModulePtr<GraphicBufferAllocator> create_buffer_allocator(
-        Display const& output) = 0;
+    virtual UniqueModulePtr<GraphicBufferAllocator> create_buffer_allocator(Display const& output) = 0;
+
+    /**
+     * Attempt to acquire a platform-specific interface from this RenderingPlatform
+     *
+     * Any given platform is not guaranteed to implement any specific interface,
+     * and the set of supported interfaces may depend on the runtime environment.
+     *
+     * Since this may result in a runtime probe the call may be costly, and the
+     * result should be saved rather than re-acquiring an interface each time.
+     *
+     * \tparam Interface
+     * \return  On success: an occupied std::shared_ptr<Interface>
+     *          On failure: std::shared_ptr<Interface>{nullptr}
+     */
+    template<typename Interface>
+    static auto acquire_interface(std::shared_ptr<RenderingPlatform> platform) -> std::shared_ptr<Interface>
```

Comment:
This seems like needless complexity: There is only one specialization of this template used:

```
auto gl_provider = mg::RenderingPlatform::acquire_interface<mg::GLRenderingProvider>(...)
```